### ISSUE TITLE
One question regarding padding

### ISSUE THREAD

**Issue by Chandrak1907** [mxhofer/Named-Entity-Recognition-BidirectionalLSTM-CNN-CoNLL#4]
hi,

I see that you are padding the inputs to get equal length of 52. But, it seems padding is applied to only character inputs but not to words.

```
 # 0-pads all words
 def padding(Sentences):
     maxlen = 52
     for sentence in Sentences:
         char = sentence[2]
         for x in char:
             maxlen = max(maxlen, len(x))
     for i, sentence in enumerate(Sentences):
         Sentences[i][2] = pad_sequences(Sentences[i][2], 52, padding='post')
     return Sentences

```

Sentences contains below:

```
         dataset.append([wordIndices, caseIndices, charIndices, labelIndices]) 
     return dataset

```

I see that you have made batches of inputs with words of equal length.  Is this the correct approach? 
Can you pls let me know.

**Comment by mxhofer** [comment]
Hello! That's correct. This is because the convolutional neural net (CNN) processes the padded character vectors of equal length. Alternatively, one could split the CNN input into equal-size batches (e.g. [here](https://stackoverflow.com/questions/46204569/how-to-handle-variable-sized-input-in-cnn-with-keras?)). Each input batch to the bi-directional LSTM has the same length, depending on how many words there are in a document.

**Comment by Chandrak1907** [comment]
Thank you for responding. One follow up question.
What was the rationale behind padding characters to a maximum length of 52. There will be 26 upper case letters, 26 lower case letters and other punctuation characters.  Can you pls let me know?

**Comment by mxhofer** [comment]
A maximum length was chosen after analyzing the longest words in the documents, such that no words are cut off.

**Comment by Chandrak1907** [comment]
There is some confusion. In my understanding, padding is applied to only **characters** not to words.

**Comment by mxhofer** [comment]
Padding is indeed applied to characters. For example, the padded character-level input below is for the word "RECORD". The output of the `padding(Sentences)` function is a list of documents of a list of words, cases, characters and labels (see output of the `createMatrices(sentences, word2Idx, label2Idx, case2Idx, char2Idx)` function.
<img width="585" alt="screenshot 2018-12-16 at 14 40 57" src="https://user-images.githubusercontent.com/16178169/50054304-c075ef80-0140-11e9-8ed9-82f4b90542ab.png">
